sdk/core/azure-core/src/main/java/com/azure/android/core/util/paging/PagedCollection.java

```diff
@@ -0,0 +1,119 @@
+package com.azure.android.core.util.paging;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public class PagedCollection<T, P extends Page<T>> implements Iterable<T> {
+    private static final int DEFAULT_PAGE_SIZE = 25; // Is this a good default size?
+
+    private PageRetriever<T, P> pageRetriever;
+    private Integer pageSize;
+    private List<T> items;
```

Comment:
One thing I was not sure about was if we needed the Page class at all. It is possible to have a list of items instead of a list of Pages and then return a slice of the `items` whenever a page is requested. Not sure what option is better.